haskell-src/Concordium/Types/Queries.hs

```diff
@@ -689,3 +698,199 @@ data EChainParametersAndKeys = forall (cpv :: ChainParametersVersion).
     { ecpParams :: !(ChainParameters' cpv),
       ecpKeys :: !(U.UpdateKeysCollection cpv)
     }
+
+instance ToJSON EChainParametersAndKeys where
+    toJSON (EChainParametersAndKeys (params :: ChainParameters' cpv) keys) =
+      case chainParametersVersion @cpv of
+        SCPV0 -> object [
+            "version" .= String "0",
+            "parameters" .= toJSON params,
+            "updateKeys" .= toJSON keys
+          ]
+        SCPV1 -> object [
+            "version" .= String "1",
+            "parameters" .= toJSON params,
+            "updateKeys" .= toJSON keys
+          ]
+
+-- |The committee information of a node which is configured with
+-- baker keys but is somehow is _not_ part of the current baking
+-- committee.
+data PassiveCommitteeInfo
+    = -- |The node is started with baker keys however it is currently not in
+      -- the baking committee. The node is _not_ baking.
+      NotInCommittee
+    | -- |The account is registered as a baker but not in the current `Epoch`.
+      -- The node is _not_ baking.
+      AddedButNotActiveInCommittee
+    | -- |The node has configured invalid baker keys i.e., the configured
+      -- baker keys do not match the current keys on the baker account.
+      -- The node is _not_ baking.
+      AddedButWrongKeys
+  deriving (Show)
+
+-- |Status of the baker configured node.
+data BakerConsensusInfoStatus
+    = -- |The node is currently not baking.
+      PassiveBaker !PassiveCommitteeInfo
+    | -- |Node is configured with baker keys and active in the current baking committee
+      ActiveBakerCommitteeInfo
+    | -- | Node is configured with baker keys and active in the current finalizer
+      -- committee (and also baking committee).
+      ActiveFinalizerCommitteeInfo
```

Comment:
These are odd names for variants.  It would make more sense to just call them "ActiveBaker" and "ActiveFinalizer".